site/docs/v1/tech/lambdas/testing-lambdas.md

```diff
@@ -0,0 +1,380 @@
+  - `/api/lambda`
+  - `/api/lambda/search`
+  - `/api/login`
+  - `/api/user`
+  - `/api/user/registration`
+
+  These endpoints are necessary to perform actions on lambdas, such as updating and deleting, and to mimic user actions to test the lambdas, such as logging in.
+- Save the API key.
+
+### Use The Lambda CLI
+
+First, install the Node CLI library. Open a terminal in your app folder and use the following commands.
+
+```bash
+npm install --save-dev @fusionauth/cli
+npx fusionauth --help
+```
+
+You should see the FusionAuth logo and a usage help message.
+
+The lambda commands that the CLI provides match operations in the underlying TypeScript client library: `update`, `delete`, and `retrieve`.
+
+Now you can retrieve the "[ATest]" lambda you created earlier. This is a useful way to check that a lambda you've created has been successfully uploaded for your app.
+
+```bash
+npx fusionauth lambda:retrieve f3b3b547-7754-452d-8729-21b50d111505 --key lambda_testing_key
+```
+
+The lambda will be saved to a file, where the file name is the UUID of your lambda. So it should look like this: `./lambdas/f3b3b547-7754-452d-8729-21b50d111505.yaml`.
+
+Let's update the lambda to say "Goodbye World!" instead of "Hello World!" and re-upload it. Open the file in a text editor, change the value of the <span>body</span>{:.field} property to the following.
+
+{% raw %}
+```
+body: |
+  function populate(jwt, user, registration) {
+    jwt.message = 'Goodbye World!';
+    console.info('Goodbye World!');
+  }
+```
+{% endraw %}
+
+Save the file and upload the lambda with the following command.
+
+```bash
+npx fusionauth lambda:update f3b3b547-7754-452d-8729-21b50d111505 --key lambda_testing_key
+```
+
+You can check that the lambda in FusionAuth now says "Goodbye World!" by viewing the ["[ATest]" lambda details](http://localhost:9011/admin/lambda).
+
+{% include docs/_image.liquid src="/assets/img/docs/customization/lambdas/testing-lambdas/updated-lambda.png" alt="Update lambda" class="img-fluid bottom-cropped" width="1200" figure=false %}
+
+### Other CLI Functionality
+
+You can create a lambda using the Node CLI with `lambda:create` and delete a lambda with `lambda:delete`, but only if the lambda is not linked to an application. To link a lambda to an application, use `lambda:link-to-application` with existing application and lambda Ids. Linking a lambda will use it for both the application's <span>Access Token populate lambda</span>{:.field} and <span>Id Token populate lambda</span>{:.field}. To unlink a lambda from an application, use `lambda:unlink-from-application` with existing application and lambda Ids. Unlinking a lambda will remove it from both the fields or just one of them if the other links to a different lambda. To link types of lambdas other than JWT populate, use the underlying client library or the admin UI.
```

Comment:
As discussed, let's replace this with a screenshot showing how to associate the lambda with an application. You can mention that this can be done via the client library or API as well.